lib/fieldhand/paginator.rb

```diff
@@ -86,6 +90,19 @@ def parse_response(query = {})
       response_parser
     end
 
+    def retryable_request(query)
+      response = request(query)
+      raise ResponseError, response unless response.is_a?(::Net::HTTPSuccess)
```

Comment:
Do we want _all_ unsuccessful responses to be retried or just ones that are non-client errors (e.g. should we retry a request if we get a 400 Bad Request)?

Reply:
Considering requests are standardised for OAI and we do not send any payload, I would keep it simple as mainly will be 500s (I have still to see a 404, and that would be a major issue on the server side as we use provided resumption tokens)